File: backend/Nexus-IMS/src/main/java/com/learn/nexus_ims/services/SubzoneService.java
package com.learn.nexus_ims.services;

import java.util.List;

import com.learn.nexus_ims.dtos.SubzoneDto;

public interface SubzoneService {
    SubzoneDto createSubzone(SubzoneDto subzoneDto);
    SubzoneDto updateSubzone(Long zoneId, SubzoneDto subzoneDto);
    void deleteSubzone(Long zoneId);
    List<SubzoneDto> getAllSubzones();
    SubzoneDto getSubzoneById(Long zoneId);
}

File: frontend/src/services/subzoneService.js
import api from './api';

export const subzoneService = {
  getAllSubzones: async () => {
    const response = await api.get('/subzones');
    return response.data;
  },

  getSubzoneById: async (id) => {
    const response = await api.get(`/subzones/${id}`);
    return response.data;
  },

  createSubzone: async (subzoneData) => {
    const response = await api.post('/subzones', subzoneData);
    return response.data;
  },

  updateSubzone: async (id, subzoneData) => {
    const response = await api.put(`/subzones/${id}`, subzoneData);
    return response.data;
  },

  deleteSubzone: async (id) => {
    await api.delete(`/subzones/${id}`);
  },
};
